test/fw/ptl/utils/pbs_testsuite.py

```diff
@@ -828,6 +829,132 @@ def init_proc_mon(self):
+                cmds_to_exec.append(["sched", "start"])
+                dmns_to_start += 1
+            if self.moms is None or len(self.moms) == 0:
+                if (pbs_conf_val["PBS_START_MOM"] != "1"):
+                    pbs_conf_val["PBS_START_MOM"] = "1"
+                    cmds_to_exec.append(["mom", "start"])
+                    dmns_to_start += 1
+            else:
+                pbs_conf_val["PBS_START_MOM"] = "0"
+            if self.comms is None or len(self.comms) == 0:
+                if (pbs_conf_val["PBS_START_COMM"] != "1"):
+                    pbs_conf_val["PBS_START_COMM"] = "1"
+                    cmds_to_exec.append(["comm", "start"])
+                    dmns_to_start += 1
+            else:
+                pbs_conf_val["PBS_START_MOM"] = "0"
+                cmds_to_exec.append(["mom", "stop"])
+
+            # Set PBS_CORE_LIMIT, PBS_SCP and PBS_SERVER
+            pbs_conf_val["PBS_CORE_LIMIT"] = "unlimited"
+            pbs_conf_val["PBS_SERVER"] = primary_server.hostname
+            if "PBS_SCP" not in pbs_conf_val:
+                scppath = self.du.which(server.hostname, "scp")
+                if scppath != "scp":
+                    pbs_conf_val["PBS_SCP"] = scppath
+
+            self.du.set_pbs_config(server.hostname, pbs_conf_filepath,
+                                   confs=pbs_conf_val)
+
+            if dmns_to_start == 4:
+                # Restart all
+                server.restart(server.hostname)
+            else:
+                for initcmd in cmds_to_exec:
+                    # start/stop the particular daemon
+                    server.pi.initd(server.hostname, initcmd[1],
+                                    pbs_conf_filepath, daemon=initcmd[0])
+
+        # Revert mom hosts' pbs.conf
+        for mom in self.moms.values():
+            start_mom = False
+            pbs_conf_filepath = self.du.get_pbs_conf_file(mom.hostname)
+            pbs_conf_val = self.du.parse_pbs_config(mom.hostname,
+                                                    pbs_conf_filepath)
+            if pbs_conf_val is None or len(pbs_conf_val) < 1:
+                raise ValueError("Could not parse pbs.conf on host %s" %
+                                 (mom.hostname))
+
+            # Set the mom start bit to 1
+            if (pbs_conf_val["PBS_START_MOM"] != "1"):
+                pbs_conf_val["PBS_START_MOM"] = "1"
+                start_mom = True
+
+            # Set PBS_CORE_LIMIT, PBS_SCP and PBS_SERVER
+            pbs_conf_val["PBS_CORE_LIMIT"] = "unlimited"
+            pbs_conf_val["PBS_SERVER"] = primary_server.hostname
+            if "PBS_SCP" not in pbs_conf_val:
+                scppath = self.du.which(mom.hostname, "scp")
+                if scppath != "scp":
+                    pbs_conf_val["PBS_SCP"] = scppath
+
+            self.du.set_pbs_config(mom.hostname, pbs_conf_filepath,
+                                   confs=pbs_conf_val)
+
+            if start_mom:
```

Comment:
If you change PBS_CORE_LIMITS or PBS_SERVER or PBS_SCP you would still want to restart mom